apps/portal/prisma/new-schema.prisma

```diff
@@ -0,0 +1,229 @@
+datasource db {
+    provider = "mongodb"
+    url      = env("MONGODB_URI")
+}
+
+generator client {
+    provider = "prisma-client-js"
+}
+
+generator typegraphql {
+    provider = "typegraphql-prisma"
+}
+
+model Account {
+    id                String  @id @default(auto()) @map("_id") @db.ObjectId
+                                                //separate data type for primary key. By default   
+                                                //id is specified by _id. ObjectId = specific for MongoDB
+    userId            String
+    type              String
+    provider          String
+    providerAccountId String
+    refresh_token     String? @db.String 
+    access_token      String? @db.String
+    expires_at        Int?
+    token_type        String?
+    scope             String?
+    id_token          String? @db.String
+    session_state     String?
+
+    user User @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+    @@unique([provider, providerAccountId])
+    @@map("accounts")
+}
+
+model EventReservation {
+    id        String  @id @default(auto()) @map("_id") @db.ObjectId
+    profile   Profile @relation(fields: [profileId], references: [id])
+    profileId String
+    event     Event   @relation(fields: [eventId], references: [id])
+    eventId   String
+    status    String
+
+    @@unique([profileId, eventId])
+    @@map("event_reservations")
+}
+
+model Session {
+    id           String   @id @default(auto()) @map("_id") @db.ObjectId
+    sessionToken String   @unique
+    userId       String
+    expires      DateTime
+    user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+    @@map("sessions")
+}
+
+model Role {
+    id       String @id @default(auto()) @map("_id") @db.ObjectId
+    roleName String
+
+    RolesOnUser RolesOnUser[]
+
+    @@map("roles")
+}
+
+model RolesOnUser {
+    id        String   @id @default(auto()) @map("_id") @db.ObjectId
+    role      Role     @relation(fields: [roleId], references: [id])
+    roleId    String
+    user      User     @relation(fields: [userId], references: [id])
+    userId    String
+    createdAt DateTime @default(now())
+
+    @@unique([userId, roleId])
+}
+
+model User {
+    id            String        @id @default(auto()) @map("_id") @db.ObjectId
+    name          String?
+    email         String?       @unique
+    emailVerified DateTime?
+    image         String?
+    accounts      Account[]
+    sessions      Session[]
+    profile       Profile?
+    roles         RolesOnUser[]
+
+    applications    UserAppJunction[]   @relation("UserLink")
```

Comment:
Feels like this would be better if hooked up with the `Profile` model instead of `User`, as the `User` model in this schema is more for the purpose of compatibility with NextAuth, which is what we use for authentication, while `Profile` is more of a source of truth of user' data for member portal.

Reply:
gotcha